### PR TITLE
add --silence-http-endpoint-warning flag

### PR DESCRIPTION
**Pull request by ubergesundheit**
Towards giantswarm/giantswarm#12975
Requires https://github.com/giantswarm/gscliauth/pull/25

This PR adds a `--silence-http-endpoint-warnings` flag which disables `Warning: endpoint URL uses an insecure protocol` messages printed when using gsctl with a http endpoint.

### CODE REVIEW

File: commands/root.go [hunk 1]
```diff
@@ -57,6 +59,7 @@ func init() {
 	RootCommand.PersistentFlags().StringVarP(&flags.Token, "auth-token", "", tokenFromEnv, "Authorization token to use")
 	RootCommand.PersistentFlags().StringVarP(&flags.ConfigDirPath, "config-dir", "", config.DefaultConfigDirPath, "Configuration directory path to use")
 	RootCommand.PersistentFlags().BoolVarP(&flags.Verbose, "verbose", "v", false, "Print more information")
+	RootCommand.PersistentFlags().BoolVarP(&flags.SilenceHTTPEndpointWarning, "silence-http-endpoint-warning", "", false, "Dont't print warnings when deliberately using an insecure http endpoint")
```
Comment (axbarsan) [inline commands/root.go:62]:
```suggestion
	RootCommand.PersistentFlags().BoolVarP(&flags.SilenceHTTPEndpointWarning, "silence-http-endpoint-warning", "", false, "Dont't print warnings when deliberately using an insecure HTTP endpoint")
```

File: commands/root.go [hunk 2]
```diff
@@ -90,7 +93,15 @@ func init() {
 // before any command is executed (see PersistentPreRunE above).
 func initConfig(cmd *cobra.Command, args []string) error {
 	fs := afero.NewOsFs()
-	err := config.Initialize(fs, flags.ConfigDirPath)
+
+	var configLogger io.Writer
+	if flags.SilenceHTTPEndpointWarning == true {
```
Comment (axbarsan) [inline commands/root.go:98]:
```suggestion
	if flags.SilenceHTTPEndpointWarning {
```

**Review by axbarsan — APPROVED** [review]
Looking good

**Comment by marians** [comment]
What was the impact of this change regarding UX? Did the output of authentication related errors or log messages change in any way?

**Comment by ubergesundheit** [comment]
All warnings and errors from `gscliauth` are silenced when the `--silence-http-endpoint-warnings` is specified.

**Pull request merged**